src/main/scala/com/mwz/sonar/scala/ScalaPlugin.scala

```diff
@@ -46,16 +48,42 @@ object Scala {
   private val FileSuffixesPropertyKey = "sonar.scala.file.suffixes"
   private val DefaultFileSuffixes = Array(".scala")
   private val ScalaVersionPropertyKey = "sonar.scala.version"
-  private val DefaultScalaVersion = ScalaVersions.Scala_2_11.toString()
+  private val DefaultScalaVersion = ScalaVersions.Scala_2_11
+  private val ScalaVersionPattern = """(\d+)\.(\d+)(?:\..+)?""".r
   private val SourcesPropertyKey = "sonar.sources"
   private val DefaultSourcesFolder = "src/main/scala"
 
-  def getScalaVersion(settings: Configuration): String =
-    settings.get(ScalaVersionPropertyKey).toOption.getOrElse(DefaultScalaVersion)
+  private val logger = Loggers.get(classOf[Scala])
+
+  def getScalaVersion(settings: Configuration): ScalaVersion = {
+    def parseVersion(s: String): Option[ScalaVersion] = s match {
+      case ScalaVersionPattern(major, minor) =>
+        for {
+          major <- major.toIntOpt
+          minor <- minor.toIntOpt
+        } yield ScalaVersion(major, minor)
+      case _ ⇒
+        None
+    }
+
+    val scalaVersion = settings
+      .get(ScalaVersionPropertyKey)
+      .toOption
+      .flatMap(parseVersion)
+      .getOrElse(DefaultScalaVersion)
+
+    // log a warning if using the default scala version
+    if (scalaVersion == DefaultScalaVersion)
```

Comment:
In this case, you would probably still show this warning even if you set the version to `2.11.0` yourself, which is a correct version - you probably want to show this warning only when you get `None` as a result of parsing the version.

Reply:
Not really, if you set `sonar.scala.version=2.11.0`, this variable will be `ScalaVersion(2, 11)` _(because we're ignoring the patch)_. Thus it **will not be equals to** `ScalaVersion(2, 11, "0")`.
I check that before pushing :+1:

Reply:
Oh yes, you're right.